src/tests/JIT/Intrinsics/TypeIntrinsics.cs

```diff
@@ -276,6 +277,30 @@ private static void IsGenericTypeTests()
         IsTrue(new ClassUsingIsGenericTypeOnT<int[]>().IsGenericTypeFromOtherGenericType());
     }
 
+    private static void GetGenericTypeDefinitionTests()
+    {
+        AreEqual(typeof(GenericEnumClass<SimpleEnum>).GetGenericTypeDefinition(), typeof(GenericEnumClass<>));
+        AreEqual(typeof(GenericEnumClass<>).GetGenericTypeDefinition(), typeof(GenericEnumClass<>));
+        AreEqual(typeof(IGenericInterface<string>).GetGenericTypeDefinition(), typeof(IGenericInterface<>));
+        AreEqual(typeof(IGenericInterface<>).GetGenericTypeDefinition(), typeof(IGenericInterface<>));
+        AreEqual(typeof(GenericStruct<string>).GetGenericTypeDefinition(), typeof(GenericStruct<>));
+        AreEqual(typeof(GenericStruct<>).GetGenericTypeDefinition(), typeof(GenericStruct<>));
+        AreEqual(typeof(SimpleEnum?).GetGenericTypeDefinition(), typeof(Nullable<>));
+        AreEqual(typeof(int?).GetGenericTypeDefinition(), typeof(Nullable<>));
+        AreEqual(typeof(IntPtr?).GetGenericTypeDefinition(), typeof(Nullable<>));
+        AreEqual(typeof(Nullable<>).GetGenericTypeDefinition(), typeof(Nullable<>));
+        AreEqual(typeof(KeyValuePair<int,string>).GetGenericTypeDefinition(), typeof(KeyValuePair<,>));
+        AreEqual(typeof(KeyValuePair<,>).GetGenericTypeDefinition(), typeof(KeyValuePair<,>));
+        AreEqual(typeof(Dictionary<int,string>).GetGenericTypeDefinition(), typeof(Dictionary<,>));
+        AreEqual(typeof(Dictionary<,>).GetGenericTypeDefinition(), typeof(Dictionary<,>));
+        AreEqual(typeof(List<string>).GetGenericTypeDefinition(), typeof(List<>));
+        AreEqual(typeof(List<>).GetGenericTypeDefinition(), typeof(List<>));
+        AreEqual(typeof(Action<>).GetGenericTypeDefinition(), typeof(Action<>));
+        AreEqual(typeof(Action<string>).GetGenericTypeDefinition(), typeof(Action<>));
+        AreEqual(typeof(Func<string, int>).GetGenericTypeDefinition(), typeof(Func<,>));
+        AreEqual(typeof(Func<,>).GetGenericTypeDefinition(), typeof(Func<,>));
```

Comment:
Can you add some test cases for _Canon? e.g. for `typeof(T)` and `typeof(GenericClass<T>)` ?

for that you need to define a separate NoInlining generic method and then call it with T being e.g. String

Reply:
Done! Both for generic cases, and for cases where it should throw `InvalidOperationException` 🙂